platforms/jvm/language-java/src/main/java/org/gradle/jvm/component/internal/JvmSoftwareComponentInternal.java

```diff
@@ -60,4 +66,12 @@ public interface JvmSoftwareComponentInternal extends SoftwareComponent {
      */
     JvmFeatureInternal getMainFeature();
 
+    /**
+     * Sets consistent resolution for this component, this ensures that the runtime classpath of the component
```

Comment:
The javadoc here does not document that this method in its current state also configures all source sets in the project to all resolve consistently with themselves. 

The more I look at this the more I think we should update this to avoid having the component configure all other source sets. I think we can do this relatively easily without needing to resort to the changes proposed by https://github.com/gradle/gradle/issues/26420 (even though I think we should go this route eventually). 

Please consider integrating this commit into your PR. I based it off this branch so you should be able to easily fast-forward your PR to this commit:

https://github.com/gradle/gradle/commit/01ca96b146387b19e45418560b57653cf2815c0e

I think this change finds a simple middle-ground between the changes in this PR and the changes proposed by the follow-up issue you wrote